diesel/src/pg/expression/operators.rs

```diff
@@ -26,3 +28,44 @@ infix_operator!(HasAnyKeyJsonb, " ?| ", backend: Pg);
 infix_operator!(HasAllKeysJsonb, " ?& ", backend: Pg);
 infix_operator!(ContainsJsonb, " @> ", backend: Pg);
 infix_operator!(IsContainedByJsonb, " <@ ", backend: Pg);
+
+#[derive(Debug, Clone, Copy, QueryId, DieselNumericOps, ValidGrouping)]
+#[doc(hidden)]
+pub struct ArrayIndex<L, R> {
+    pub(crate) left: L,
+    pub(crate) right: R,
```

Comment:
I would prefer other names than `left` and `right` here.

```suggestion
    pub(crate) array_expr: L,
    pub(crate) index_expr: R,
```